pkg/client/kubernetes/scaling.go

```diff
@@ -120,3 +121,30 @@ func ScaleStatefulSetAndWaitUntilScaled(ctx context.Context, c client.Client, ke
 	}
 	return WaitUntilStatefulSetScaledToDesiredReplicas(ctx, c, key, replicas)
 }
+
+// WaitUntilNoPodsForDeployment waits for all pods matching the deployment's selector to be terminated.
+func WaitUntilNoPodsForDeployment(ctx context.Context, c client.Client, key client.ObjectKey, interval, timeout time.Duration) error {
```

Comment:
I just noticed that we already have a generic implementation for this: https://github.com/gardener/gardener/blob/fc1fdfcfb9f029a88005ec3f98178c68d79f49f7/pkg/utils/kubernetes/kubernetes.go#L164-L186

Reply:
This function does not provide a timeout parameter. The function cancels with the passed context. Should we use it or instead leave the new more concrete implementation?

Reply:
Typically, it is the responsibility of the caller to pass a context with a proper timeout, e.g. like in https://github.com/gardener/gardener/blob/bc1827317538914af9be133cba42dfe7bb806e24/pkg/operation/botanist/component/clusterautoscaler/bootstrap.go#L85-L88

Reply:
I reused the generic function and removed the concrete implementation of the `scaling` package.